core/src/main/java/org/apache/brooklyn/util/core/task/BasicExecutionManager.java

```diff
@@ -797,6 +800,7 @@ public TaskLookup(BasicExecutionManager mgr, Task<?> t) {
                 .description("Details of the original task have been forgotten.")
                 .body(Callables.returning((T)null)).build();
             ((BasicTask<T>)t).ignoreIfNotRun();
+            ((BasicTask<T>)t).cancelled = true;
```

Comment:
Worth a comment saying why you're setting `cancelled = true`. I presume it's because we don't really want anyone executing the "gone" task, and more importantly that if we are GC'ing tasks then marking the "gone" task as cancelled will help with cleanup of sub-tasks that have lost their submitted-by-task reference?

Reply:
never occurred to me that marking cancelled might help with GC.  have added comments.

the only thing i was optimizing was the surprising expense of soft references (noted 30 lines earlier)